maven-core/src/main/java/org/apache/maven/lifecycle/internal/builder/BuilderCommon.java

```diff
@@ -105,6 +106,20 @@ public MavenExecutionPlan resolveBuildPlan( MavenSession session, MavenProject p
 
         lifecycleDebugLogger.debugProjectPlan( project, executionPlan );
 
+        // With Maven 4's build/consumer the POM will always rewrite during distribution.
+        // The maven-gpg-plugin uses the original POM, causing an invalid signature.
+        // Fail as long as there's no solution available yet
+        Optional<MojoExecution> gpgMojo = executionPlan.getMojoExecutions().stream()
+                .filter( m -> "org.apache.maven.plugins".equals( m.getGroupId() ) )
```

Comment:
Can't the filter be ANDed?

Reply:
Sure it can, but IMO this is more readable. The effect is the same.

Reply:
Think of it like `if (.. && ..)`.